libgalois/include/galois/worklists/AdaptiveObim.h

```diff
@@ -473,8 +471,7 @@ struct AdaptiveOrderedByIntegerMetric : private boost::noncopyable {
 
   void push(const value_type& val) {
     ThreadData& p = *current.getLocal();
-    while (!p.lock.try_lock())
```

Comment:
The explicit lock here is fine. I prefer to start a new issue/pr for replacing some explicit lock with std::lock_guard if necessary.